services/mgmt/alertsmanagement/src/package_2018_05/models.rs

```diff
@@ -83,13 +83,13 @@ pub struct EgressConfig {}
 pub struct AlertContext {}
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
 pub struct Essentials {
-    #[serde(skip_serializing)]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
```

Comment:
More of a general comment: I'm surprised serde doesn't do this for `Option<T>` in general like serializers do in our other languages. Is there some way we can modify the behavior and not have to attribute every optional or vectored field?

Reply:
The reason would be that some serialization formats require all fields to be present. `serde` is for more than just json so even if you could argue that for json the field should just not be present (which I don't think is a thing one could claim), it might not work for other seralization formats.